spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -119,7 +119,7 @@ public class SparkCatalog extends BaseCatalog {
   protected Catalog buildIcebergCatalog(String name, CaseInsensitiveStringMap options) {
     Configuration conf = SparkUtil.hadoopConfCatalogOverrides(SparkSession.active(), name);
     Map<String, String> optionsMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
-    optionsMap.putAll(options);
+    optionsMap.putAll(options.asCaseSensitiveMap());
```

Comment:
Why is this needed? The `optionsMap` is a treemap that is case insensitive.

Reply:
This is needed to save case sensitive [S3 write/delete tag keys](https://github.com/apache/iceberg/blob/master/docs/integrations/aws.md#s3-tags) for objects.

There is a thread in `general` channel on this, let me add you there: https://apache-iceberg.slack.com/archives/C025PH0G1D4/p1653879105452189

I have removed the case insensitive comparator from the Treemap as well.

Reply:
Got it. So this passes the original case for when we extract the keys. Thanks!

Reply:
Yes, correct! 

Thanks, for merging. I'll backport the changes for Spark 3.x.

Reply:
I have raised the backport PR for Spark 3.x: https://github.com/apache/iceberg/pull/5044